File: datalib/src/main/java/nl/acidcats/tumblrlikes/core/usecases/appsetup/TumblrBlogUseCaseImpl.java
package nl.acidcats.tumblrlikes.core.usecases.appsetup;

import javax.inject.Inject;

import nl.acidcats.tumblrlikes.core.repositories.AppDataRepository;
import rx.Observable;

/**
 * Created on 17/10/2018.
 */
public class TumblrBlogUseCaseImpl implements TumblrBlogUseCase {

    private AppDataRepository _appDataRepository;

    @Inject
    public TumblrBlogUseCaseImpl(AppDataRepository appDataRepository) {
        _appDataRepository = appDataRepository;
    }

    @Override
    public Observable<Boolean> setTumblrBlog(String tumblrBlog) {
        _appDataRepository.setTumblrBlog(tumblrBlog);

        return Observable.just(true);
    }

    @Override
    public Observable<String> getTumblrBlog() {
        return Observable.just(_appDataRepository.getTumblrBlog());
    }
}

File: datalib/src/main/java/nl/acidcats/tumblrlikes/core/usecases/appsetup/TumblrBlogUseCaseImpl.kt
package nl.acidcats.tumblrlikes.core.usecases.appsetup

import nl.acidcats.tumblrlikes.core.repositories.AppDataRepository
import rx.Observable
import javax.inject.Inject

/**
 * Created on 24/10/2018.
 */
class TumblrBlogUseCaseImpl @Inject constructor(val appDataRepository: AppDataRepository) : TumblrBlogUseCase {
    override fun getTumblrBlog(): Observable<String> {
        return Observable.just(appDataRepository.getTumblrBlog())
    }

    override fun setTumblrBlog(tumblrBlog: String): Observable<Boolean> {
        appDataRepository.setTumblrBlog(tumblrBlog)

        return Observable.just(true)
    }
}